### PR TITLE
Disable LTO for ASAN builds

### PR DESCRIPTION
**Pull request by aandvalenzuela**
To test if new ASAN errors are related to LTO, as requested in https://github.com/cms-sw/cmssw/issues/43003.

### CODE REVIEW

File: compilation_flags_lto.file [hunk 1]
```diff
@@ -1,11 +1,19 @@
...
 %undefine enable_lto
 %define enable_lto 0
 %endif
+
+#Disable LTO for ASAN
+%if "%(case %realversion in (*_ASAN_X*) echo true ;; (*) echo false ;; esac)" == "true"
+%undefine enable_lto
+%define enable_lto 0
+%endif
+
```
Comment (smuzaffar) [inline compilation_flags_lto.file]: @aandvalenzuela , this is not going to work. this file is not directly used by cmssw so `realversion` will not container `ASAN`. I would suggest to just change `%define enable_lto 1` to `%define enable_lto 0` on the 2nd line of this file and try to build full cmssw for ASAN

Comment (aandvalenzuela) [reply compilation_flags_lto.file]: Ah, nice! In fact, I was a bit unsure of how to filter based on flavor in this file. Do I make the change, trigger the IB, and revert the change?

Comment (smuzaffar) [reply compilation_flags_lto.file]: No, no need to build the IB. Just the PR tests should be enough to see if LTO is causing any issues for ASAN

**Comment by aandvalenzuela** [comment]
please test for CMSSW_13_3_ASAN_X
- full_cmssw = true

**Comment by smuzaffar** [comment]
test parameters:
- full_cmssw = true

**Comment by smuzaffar** [comment]
> please test for CMSSW_13_3_ASAN_X
> 
>     * full_cmssw = true

this is not going to work. `full_cmssw = true` should be under `test parameters`

**Comment by smuzaffar** [comment]
please test for CMSSW_13_3_ASAN_X

**Comment by aandvalenzuela** [comment]
> > please test for CMSSW_13_3_ASAN_X
> > ```
> > * full_cmssw = true
> > ```
> 
> this is not going to work. `full_cmssw = true` should be under `test parameters`

My bad, sorry!

**Pull request closed without merging**